pkg/network/passt/repair_test.go

```diff
@@ -0,0 +1,211 @@
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/network/passt"
+)
+
+var _ = Describe("Passt Repair Handler", func() {
+	stubSocketDirFunc := func(_ *v1.VirtualMachineInstance) (string, error) {
+		return "/var/run/passt", nil
+	}
+
+	DescribeTable("Should not run passt repair", func(vmi *v1.VirtualMachineInstance) {
+		clusterConfig := stubClusterConfig{
+			registeredPlugins: map[string]v1.InterfaceBindingPlugin{
+				"passt": {
+					SidecarImage:                "passt:latest",
+					NetworkAttachmentDefinition: "default/passt-network",
+				},
+				"tap": {
+					DomainAttachmentType: v1.Tap,
+				},
+				"managedTap": {
+					DomainAttachmentType: v1.ManagedTap,
+				},
+			},
+		}
+
+		handler := passt.NewRepairManagerWithOptions(
+			clusterConfig,
+			fakeCreateShortenedSymlink,
+			fakeFindRepairSocketInDir,
+			fakeCommand,
+			newStubStore(),
+		)
+
+		Expect(handler.HandleMigrationSource(vmi, stubSocketDirFunc)).To(Succeed())
+	},
+		Entry("When there are no networks",
+			libvmi.New(libvmi.WithAutoAttachPodInterface(false)),
+		),
+		Entry("When an iface connected to pod network uses masquerade binding",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When an iface connected to pod network uses bridge binding",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding("default")),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When an iface connected to pod network uses tap attachment",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceWithBindingPlugin("default", v1.PluginBinding{Name: "tap"})),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When an iface connected to pod network uses managedTap attachment",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceWithBindingPlugin("default", v1.PluginBinding{Name: "managedTap"})),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			),
+		),
+		Entry("When there is no iface connected to pod network",
+			libvmi.New(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding("secondary")),
+				libvmi.WithNetwork(libvmi.MultusNetwork("secondary", "secondary-nad")),
+			),
+		),
+	)
+
+	DescribeTable("Should run passt repair", func(vmi *v1.VirtualMachineInstance) {
+		clusterConfig := stubClusterConfig{
+			registeredPlugins: map[string]v1.InterfaceBindingPlugin{
+				"passt": {
+					SidecarImage:                "passt:latest",
+					NetworkAttachmentDefinition: "default/passt-network",
+				},
+			},
+		}
+
+		handler := passt.NewRepairManagerWithOptions(
+			clusterConfig,
+			fakeCreateShortenedSymlink,
+			fakeFindRepairSocketInDir,
+			fakeCommand,
+			newStubStore(),
+		)
+		Expect(handler.HandleMigrationSource(vmi, stubSocketDirFunc)).To(Succeed())
```

Comment:
IMO the test's main goal should be to assert that the (stub) "passt-repair" command was executed.